### DISCUSSION TITLE
BS5 templates are broken

### DISCUSSION THREAD

**Discussion by darylknight** [solspace/craft-freeform#1269] [Questions & Answers] [answered]
**Describe the bug or issue you're experiencing**
There are multiple other issues with the Bootstrap 5 template, migrating from Freeform 4 to 5
**Steps to reproduce**

1. The labels have been moved underneath the inputs
2. The `form-group` class has been removed
3. `form-floating` has been added in it's place

**Expected behavior**
Templates should be the same as before

Freeform 4:
![image](https://github.com/solspace/craft-freeform/assets/1250017/43abc74f-c443-4ecf-a02f-83263c7fd941)
![image](https://github.com/solspace/craft-freeform/assets/1250017/7520fb6c-e093-40b3-87c8-ec6b982888e9)

Freeform 5:
![image](https://github.com/solspace/craft-freeform/assets/1250017/868a229d-effa-4920-87ca-b9118a9b0ecc)
![image](https://github.com/solspace/craft-freeform/assets/1250017/a6005bf7-5de0-4874-a73d-61b5a58734b4)

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 4.9.1
- Freeform Version: 5.x
- Freeform Edition: Pro
- Fresh Install or Upgrade: Upgrade

**Comment by kjmartens** [comment]
Hi @darylknight,

I agree that the sample formatting templates should stay the same as much as possible for those who rely on them, but Freeform 5 was a major version change, and many parts were completely rewritten.

Due to large improvements with changes to formatting templates, each one, while fairly similar to before, was rewritten. During this time, we would have also implemented better practices and/or have them take advantage of the latest version of the framework (in this case Bootstrsap 5.3), whereas they might have used older practices from x.0.x. 🙂 

That said, I am not 100% sure which formatting template you're referring to. Are you using the **Bootstrap 5 Floating Labels** sample template? If so, the "labels underneath the inputs" and `form-floating` instead of `form-group` class haven't changed between FF4 and FF5, and it's per the Bootstrap docs (where they have dropped support for `form-group`):

### Bootstrap Docs:
- https://getbootstrap.com/docs/5.3/migration/#forms-1
- https://getbootstrap.com/docs/5.3/forms/floating-labels/

### _Bootstrap 5 Floating Labels_ Formatting Template Docs for Freeform:
- Freeform 5: https://docs.solspace.com/craft/freeform/v5/templates/formatting/bootstrap-5-floating-labels/
- Freeform 4: https://docs.solspace.com/craft/freeform/v4/templates/formatting/bootstrap-5-floating-labels/

And finally, you are of course welcome to create a custom formatting template if you wish. In this case, you can support any legacy classes and formatting that are necessary for your project. 🙂 

If I have misunderstood something in your request, please let me know and I will be happy to help. 🙂 

If you mean to use the regular Bootstrap 5 formatting template, please ensure that _Bootstrap 5_ is assigned to the form instead of _Bootstrap 5 Floating Labels_... it's possible that is accidentally assigned in the form builder.

**Comment by darylknight** [comment]
Sorry about that Kelsey - I must have changed it to floating labels while I was trying to get it to work (I'm sure Bootstrap 5 template looked broken the first time). Ok, so it's mostly the same. Looks like the only changes worth noting are:

1. Placeholder colour is now grey instead of white (changing Bootstrap's `$input-placeholder-color` no longer affects this
2. Freeform appears to be adding a `placeholder-red` class to the input, although that doesn't seem to be having any effect
3. A downward chevron has now appeared in the select dropdown
4. The text of the Submit button (which used to say "Get in touch" is not migrated during the update to Freeform 5 and now says "Submit" again

![image](https://github.com/solspace/craft-freeform/assets/1250017/eb1a7946-f77e-42b3-b83d-8624b4a39160)

**Reply by kjmartens** [reply, marked as answer]
Thanks for clarifying @darylknight. 🙂 

1. This was an override we made in the CSS as it seemed too similar to the color of regular text. However, in order for it to work, we needed to use `!important`, which then means its very likely it'll be impossible for anyone else to override the color. For this reason, we'll remove that override CSS.
2. This is a bug and we'll remove it for the next release.
3. This is actually a longstanding issue that we corrected, as the chevron is supposed to be there. You can check the Bootstrap docs: https://getbootstrap.com/docs/5.3/forms/select/
   - If you need to override them, you can try these [SASS vars](https://getbootstrap.com/docs/5.3/forms/select/#sass-variables):
      - `$form-select-indicator-color: $gray-800;`
      - `$form-select-indicator: url("data:image/svg+xml,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16'><path fill='none' stroke='#{$form-select-indicator-color}' stroke-linecap='round' stroke-linejoin='round' stroke-width='2' d='m2 5 6 6 6-6'/></svg>");`
4. The migration does not cover submit button wording at this time. You'll just need to adjust it after the migration. 🙂

**Comment by darylknight** [comment]
Got it, thank you for explaining everything :)

**Reply by kjmartens** [reply]
Btw, we have just released Freeform 5.1.19.1 now, which takes care of those 2 changes. 🙂
